replication/binlogsyncer.go

```diff
@@ -120,6 +120,8 @@ type BinlogSyncerConfig struct {
 	Dialer client.Dialer
 
 	RowsEventDecodeFunc func(*RowsEvent, []byte) error
+
+	DiscardGTIDSet bool
```

Comment:
I think the DiscardGTIDSet should also skip maintaining `prevGset` and `currGset`, only skipping output is no benefit to performance

Reply:
prevGset and currGset need to maintain for reconnect. Skip output reduce the clone operation